paas-types/proto/paas.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package paas_types;
+
+service ProcessService {
+    rpc Exec (ExecRequest) returns (ExecResponse) {}
+    rpc GetLogs (LogsRequest) returns (stream LogsResponse) {}
+    rpc GetStatus (StatusRequest) returns (StatusResponse) {}
+    rpc Stop (StopRequest) returns (StopResponse) {}
+}
+
+message Uuid {
+    bytes id = 1;
+}
+
+message ExecRequest {
+    repeated string args = 1;
+}
+
+message ExecResponse {
+    Uuid id = 1;
+}
+
+message LogsRequest {
+    Uuid id = 1;
+}
+
+message LogsResponse {
+    repeated bytes lines = 1;
+}
+
+message StatusRequest {
+    Uuid id = 1;
+}
+
+message StatusResponse {
+    oneof exit_status {
+        int32 code = 1;
+        int32 signal = 2;
+    }
+}
```

Comment:
How would you distinguish exited with no error from is still running?

Reply:
Also, what other fields do you feel would be interesting to see status?

Reply:
If I'm not mistaken, the entire `oneof` structure is treated like an optional field (well, like any field in `proto3`). The difference is that setting e.g. `code`, even to a default value (0) in this case would still set the variant of the sum type to `code` (via [docs](https://developers.google.com/protocol-buffers/docs/proto3#oneof)):
> If you set a oneof field to the default value (such as setting an int32 oneof field to 0), the "case" of that oneof field will be set, and the value will be serialized on the wire.
In other words, `oneof`-s seem to be special in that "default value" vs "no value" are distinguished.

So, if *neither* `code` *nor* `signal` is set, this indicates that the process is still running. Talking in Rust terms, this structure is equivalent to

```rust
pub struct StatusResponse {
    pub exit_status: Option<ExitStatus>,
}
```

So, `None` is "running" and `Some(_)` is "exited".

---

As for other fields, perhaps consumed CPU time, peak or current RSS and similar statistics would be interesting.

Reply:
> If I'm not mistaken, the entire oneof structure is treated like an optional field

Got it, my misunderstanding then.

> As for other fields, perhaps consumed CPU time, peak or current RSS and similar statistics would be interesting.

These are interesting, other fields that are interesting would be how long the process ran, when it start, when it ended, etc.